### PR TITLE
fix: execute the gather variables task even with specific tags

### PR DESCRIPTION
**Pull request by btravouillon**
For each role, the task "include_vars ░ Gather OS specific variables" **MUST** be executed whatever the selection of tags.

Otherwise, other tasks in the role will fail because required variables are not defined.

Example (before this change):
```
# ansible-playbook playbooks/managements.yml -l management1 --tags firewall
[WARNING]: Unable to parse /etc/bluebanquise/internal as an inventory source

PLAY [managements playbook] ******************************************************

TASK [Gathering Facts] ***************************************************************
Tuesday 20 October 2020  19:22:09 -0400 (0:00:00.067)       0:00:00.067 *******
ok: [management1]

PLAY [managements playbook] ******************************************************
TASK [Gathering Facts] ***************************************************************
Tuesday 20 October 2020  19:22:11 -0400 (0:00:02.622)       0:00:02.689 *******
ok: [management1]

TASK [repositories_server : firewalld █ "Add services to firewall's public zone"] *
Tuesday 20 October 2020  19:22:13 -0400 (0:00:01.753)       0:00:04.443 *******
fatal: [management1]: FAILED! => 
  msg: '''repositories_server_firewall_services_to_add'' is undefined'

PLAY RECAP **************************************************************************
management1 : ok=2 changed=0 unreachable=0 failed=1 skipped=0 rescued=0 ignored=0
```

Indeed, `repositories_server_firewall_services_to_add` is not defined since the role did not gather the variables.

I use this PR to add coherency between roles: use same task name, same tags, etc.

### CODE REVIEW

**Comment by oxedions** [comment]
Dear @btravouillon 

Thank you for this PR 🙂 
You are right, this is an issue that must be solved.

I think in my side that I would prefer to add tags these include are related of.

For example in a role containing packages, services, and firewall related tasks (tagged) then I would add tags package, service and firewall.

Having always for each of these includes may result in some kind of overhead when executing only a single role (which happen much more time on my side than executing the whole playbook or a specific task tag).

**Comment by bouriquet** [comment]
> 
> 
> Dear @btravouillon
> 
> Thank you for this PR 🙂
> You are right, this is an issue that must be solved.
> 
> I think in my side that I would prefer to add tags these include are related of.
> 
> For example in a role containing packages, services, and firewall related tasks (tagged) then I would add tags package, service and firewall.
> 
> Having always for each of these includes may result in some kind of overhead when executing only a single role (which happen much more time on my side than executing the whole playbook or a specific task tag).

Agree with this solution

**Review by hmeAtos — APPROVED** [review]
ok for me @btravouillon 
thanks

**Comment by oxedions** [comment]
@btravouillon Do you want me to update the PR with this solution ?

**Comment by btravouillon** [comment]
No I will.

**Comment by btravouillon** [comment]
I wrote a quick & dirty helper script yesterday evening to ease the maintenance of these tags.

For each role, the logic is:

1. Search for any variable in `vars/*.yml`
2. For each variable, search any usage in templates or tasks
3. If found, collect the tags of the relevant tasks (including the template task if any)
4. Compare the tags with the list of tags of the first task "include_vars ░ Gather OS specific variables"
5. Report an error if some tags are missing for include_vars.

I added this script to the static code analysis in the CI, current result below: https://github.com/bluebanquise/bluebanquise/pull/434/checks?check_run_id=1295868755

I may have forgotten some use cases, feel free to comment.

File: roles/core/bluebanquise/tasks/main.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@
       skip: true
   tags:
     - internal
+    - always
```
Comment (btravouillon) [inline roles/core/bluebanquise/tasks/main.yml:23]: not detected

File: roles/core/dns_client/tasks/main.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@
       skip: true
   tags:
     - internal
+    - always
```
Comment (btravouillon) [inline roles/core/dns_client/tasks/main.yml:23]: not detected

**Comment by btravouillon** [comment]
BTW, can we remove (or rename) the file [roles/core/pxe_stack/tasks/Ubuntu_18.yml](https://github.com/bluebanquise/bluebanquise/blob/master/roles/core/pxe_stack/tasks/Ubuntu_18.yml)?

**Comment by oxedions** [comment]
You can remove it. Ubuntu 18.04 should not be supported any more, and the 20.04 is a mess to deploy, so no way currently to use it.

File: roles/core/bluebanquise/tasks/main.yml (hunk 1 above)
Comment (bouriquet) [reply roles/core/bluebanquise/tasks/main.yml:23]: why `always ` instead of `package ` tag ?

Comment (btravouillon) [reply roles/core/bluebanquise/tasks/main.yml:23]: The helper script gather_vars_checker.py did not detect the correct tags to use. I made a mistake in the script, I'm working on the fix at this exact moment :grin: .

**Review by bouriquet — APPROVED** [review]

**Comment by oxedions** [comment]
Dear @btravouillon 

Thanks a lot for this PR. I will need few days to review it entirely with the python code. 🙂

**Review by oxedions — APPROVED** [review]
Dear @btravouillon 

Thank you for this PR.
Ok for me, merging.

**Pull request merged**